README.md

```diff
@@ -1,17 +1,41 @@
 # ImageProcessing
 
-Provides higher-level image processing functionality that is commonly needed
-when accepting user uploads. Supports processing with [libvips] and [ImageMagick].
+Provides higher-level image processing helpers that are commonly needed
+when handling image uploads.
+
+This gem can process images with either [libvips] and [ImageMagick] libraries.
+ImageMagick is a good default choice, especially if you are migrating from
+another attachment gem that uses ImageMagick. Livips is a very fast alternative
+that works best with JPEGs and PNGs. Support for GIFs is limited in libvips.
+
+
+## Goal
+
+The goal of this project is to have a single gem that contains all the
+helper methods needed to resize and process images. Currently, existing
+attachment gems (like Paperclip, CarrierWave, Refile, Dragonfly, 
+ActiveStorage, and others) implement their own custom image helper methods.
+But why?
+
+Let's be honest. Image processing is a dark, mysterious art. So we want to 
+combine everything bit of best practice from all of these gems into a single, 
+awesome library that is constantly updated with best-practice thinking about 
+how to resize and process images.
 
-The goal of this project is to have a single place where common image
-processing helper methods are maintained, instead of Paperclip, CarrierWave,
-Refile, Dragonfly and ActiveStorage each implementing their own versions.
 
 ## Installation
 
-```rb
-gem "image_processing", "~> 0.11"
-```
+1. Install ImageMagick and Libvips:
+
+`$ brew install imagemagick vips`
+
+Note: if you're not on macOS or don't want to use Homebrew, check the project
+pages for ImageMagick and Libvips for other ways to install these libraries.
+
+2. Add the gem to your Gemfile:
+
+`gem 'image_processing', '~> 0.11'`
+
```

Comment:
Maybe instead of ImageMagick/libvips installation instructions, we could link here to the "backend" documentations (`vips.md` and `minimagick.md`), as they show which versions of `mini_magick` and `ruby-vips` gems should be installed, and they also link to the installation documentations for ImageMagick and libvips. That way we can avoid repeating ourselves.

Also, the proposed change makes it appear that ImageProcessing needs *both* ImageMagick and libvips installed.

Reply:
I think duplication is okay in this case.

People should be able to dive into using image_processing just by reading the README. If we don't say how to install ImageMagick and/or libvips there, that will be a potential point of confusion. ("Well, I installed the gem, but it's not working.)

So, I think we should the basic library install in the README, and then supplement that with more complete documentation in the docs for each related library.

For instance, when installing vips, which option should people use?

`brew install vips`

`brew install vips --with-imagemagick`

libvips docs say:

> That will install vips with most optional add-ons included. It won’t include ImageMagick/GraphicsMagick support (used for loading things like BMP), you have to enable that if you need it

But in the context of image_processing, I'm not entirely sure what it means. I think it means that libvips can shell out to ImageMagick if needed, but that image_processing users who wish to use the minimagick implementation will still need to install ImageMagick separately, right?

That's the kind of thing we could mention in the individual docs, but still keep simple, high-level dependency installation instructions in the README.

Reply:
I also think, given the ease of uninstalling Homebrew packages, having a single command is fine, and it will allow folks to quickly experiment with both image library APIs.

Reply:
You make a good point. Ok, let's keep the installation in the README then. We should then probably also instruct people to add the `mini_magick`/`ruby-vips` gems in the Gemfile.

Reply:
Actually, what about including both gems as dependencies?

I realize why they were kept separate, but it's sort of an annoying extra installation step. The gems are tiny too. The mini_magick gem is 43KB and the ruby-vips gem is 233KB. I might feel differently if they were each multiple megabytes in size.

Why not just include them both in image_processing? It would make install even easier.

Reply:
At first I was reluctant even after your suggestion, because I know that people don't like seeing extra dependencies (even if they're not loaded). But man, that would make keeping up-to-date with `mini_magick` and `ruby-vips` gems so much easier, because we would be able to bump the dependencies. I think that justifies depending on both gems even though most likely only one will be used.

To avoid mixing "responsibilities", I'll make that change after this PR is merged.

Reply:
Awesome! 👍

I'll push related updates to the docs now.

Reply:
Pushed!

Reply:
Great! So the PR is good to merge?

Reply:
I think so, yes!